src/pudl/analysis/plant_parts_eia.py

```diff
@@ -1483,10 +1483,14 @@ def match_to_single_plant_part(
     """
     # select only the plant-part records that we are trying to scale to
     ppe_part_df = ppe[ppe.plant_part == part_name]
+    if not pd.api.types.is_datetime64_any_dtype(ppe_part_df["report_date"]):
+        ppe_part_df["report_date"] = pd.to_datetime(ppe_part_df["report_date"])
+    if not pd.api.types.is_datetime64_any_dtype(multi_gran_df["report_date"]):
+        multi_gran_df["report_date"] = pd.to_datetime(multi_gran_df["report_date"])
```

Comment:
This is kind of janky but I think when inputs are read in from S3 (instead of from Dagster storage) the report date is an object not a datetime. I was getting errors, so I threw this in there, but it's a little sloppy.

Reply:
Should this be in the python module? Or a pre-processing step in the notebook? If the dataframe doesn't match the expected schema, then maybe it should fail?

When we read from Dagster storage if it's not a persisted asset it'll be a pickled dataframe, with the datetime type intact. If we're reading from a persisted asset, it'll be coming from Parquet, and IIRC our IOManager will enforce the correct PUDL dtypes when we read the data in. If you're reading a Parquet file from S3 and passing it in outside of the Dagster context, you might try using `dtype_backend="pyarrow"` to ensure that the datetime column gets parsed not as an object.

Reply:
Ah okay let me try using the pyarrow backend.

Reply:
The pyarrow backend introduced other type issues, so I updated to cast to a datetime within the notebook, after reading in the Parquet inputs.